CONTRIBUTING.md

```diff
@@ -16,7 +25,145 @@ Please note the below! We appreciate everyone following the guidelines.
 * Use non-master branches, using a short meaningful description, with words separated by dash (e.g. 'fix-this-bug');
 * All modifications must be made in a pull-request to solicit feedback from other contributors.
 
-## Get in Touch
+## <a name="commit"></a> Commit Message Format
+
+We have very precise rules over how our Git commit messages must be formatted, following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) guideline.
+This format leads to **easier to read commit history**.
+
+
+Each commit message consists of a **header**, a **body**, and a **footer**.
+
+
+```
+<header>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The `header` is mandatory and must conform to the [Commit Message Header](#commit-header) format.
+
+The `body` is optional.
+When the body is present it must be at least 20 characters long and must conform to the [Commit Message Body](#commit-body) format.
+
+The `footer` is optional. The [Commit Message Footer](#commit-footer) format describes what the footer is used for and the structure it must have.
+
+
+### <a name="commit-header"></a>Commit Message Header
+
+```
+<type>(<scope>): <subject>
+  │       │             │
+  │       │             └─⫸ Summary in present tense. Not capitalized. No period at the end.
+  │       │
+  │       └─⫸ Commit Scope: advance-runner|dispatcher|graphql-server|indexer|inspect-server|
+  │                          state-server|auxiliary|contracts|deploy|arbitration|cli
+  │
+  └─⫸ Commit Type: build|feat|fix|test|perf|refactor|docs|style
+```
+
+The `<type>` and `<summary>` fields are mandatory, the `(<scope>)` field is optional.
+
+
+#### Type
+
+Must be one of the following:
+
+* **build**: Changes that affect the build system or external dependencies (example: npm)
+* **feat**: A new feature
+* **fix**: A bug fix
+* **test**: Adding missing tests or correcting existing tests
+* **perf**: A code change that improves performance
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **docs**: Documentation only changes
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+
+When a commit fits multiple types, the list also serves as hierarchy of what type should be used, e.g. if a commit introduces a feature and its tests, it's a `feat`.
+
+
+#### Scope
+
+The scope has the following rules:
+
+* If you're changing the `offchain`, use the name of the crate your changes are affecting
+* If you're changing the `onchain`, use `contracts` for `rollups/contracts`; Use `deploy` for `rollups/deploy`; Use `arbitration` for `rollups-arbitration`; Use `cli` for `rollups-cli`
+
+The following is the list of supported scopes:
+
+* `advance-runner`
+* `dispatcher`
+* `graphql-server`
+* `indexer`
+* `inspect-server`
+* `state-server`
```

Comment:
It is not clear that this list is for onchain scopes. Maybe we should split the list in two.